python_modules/dagster/dagster_tests/core_tests/test_instance.py

```diff
@@ -86,3 +91,13 @@ def noop_pipeline():
 
     assert run.execution_plan_snapshot_id == ep_snapshot_id
     assert run.execution_plan_snapshot_id == create_execution_plan_snapshot_id(ep_snapshot)
+
+
+def test_dagster_home_raises(monkeypatch):
+    dirname = '.'
+    monkeypatch.setenv('DAGSTER_HOME', dirname)
```

Comment:
the approach we use in our tests is a context manager test utility called `environ` - example here https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-celery/dagster_celery_tests/test_config.py#L55